lib/join_dependency.rb

```diff
@@ -36,22 +36,32 @@ def build(relation, buckets)
       stashed_association_joins = buckets[:stashed_join]
       join_nodes                = buckets[:join_node].uniq
       string_joins              = buckets[:string_join].map(&:strip).uniq
-
+      
+      joins = string_joins.map! do |join|
```

Comment:
You're mutating the `string_joins` here, and I don't think it's intentional since you've assigned the result to `joins`.

Did you mean to use `map` here, instead of `map!`?